### PR TITLE
AON 4.2 - Initial Data Tables and Example Fix

### PR DESCRIPTION
**Pull request by TreyE**
https://www.pivotaltracker.com/story/show/185574409

### CODE REVIEW

**Review by saipraveen18 — APPROVED** [review]

File: app/models/effective/datatables/broker_agency_datatable.rb [hunk 1]
```diff
@@ -59,6 +59,10 @@ def nested_filter_definition
         }
 
       end
+
+      def authorized?(current_user, _controller, _action, _resource)
+        current_user.has_hbx_staff_role?
```
Comment (saikumar9) [inline app/models/effective/datatables/broker_agency_datatable.rb:64]:
```suggestion
        current_user&.person&.hbx_staff_role.present?
```

@TreyE Can you check if the instance of hbx_staff_role exists? The current method first checks if there is a string in an array which is the deprecated way of checking if a user has a role.

File: app/models/effective/datatables/family_data_table.rb [hunk 2]
```diff
@@ -181,6 +181,10 @@ def nested_filter_definition
           top_scope: :families
         }
       end
+
+      def authorized?(current_user, _controller, _action, _resource)
+        current_user.has_hbx_staff_role?
```
Comment (saikumar9) [inline app/models/effective/datatables/family_data_table.rb:186]: Same here

File: app/models/effective/datatables/outstanding_verification_data_table.rb [hunk 3]
```diff
@@ -187,6 +187,10 @@ def nested_filter_definition
           top_scope: :documents_uploaded
         }
       end
+
+      def authorized?(current_user, _controller, _action, _resource)
+        current_user.has_hbx_staff_role?
```
Comment (saikumar9) [inline app/models/effective/datatables/outstanding_verification_data_table.rb:192]: Same here

File: spec/models/effective/datatable/sep_type_data_table_spec.rb [hunk 4]
```diff
@@ -105,3 +105,31 @@
   end
 end
 
+describe Effective::Datatables::SepTypeDataTable, "with correct access permissions" do
```
Comment (saikumar9) [inline spec/models/effective/datatable/sep_type_data_table_spec.rb:108]: Good test coverage.

**Review by saikumar9 — CHANGES_REQUESTED** [review]

**Review by saikumar9 — APPROVED** [review]

**Pull request merged**